modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -1,57 +1,506 @@
-import { BaseTransactionBuilder } from '../baseCoin';
-import { NotImplementedError } from '../baseCoin/errors';
+import { BaseCoin as CoinConfig, DotNetwork } from '@bitgo/statics';
+import { BaseTransactionBuilder, TransactionType } from '../baseCoin';
+import { InvalidTransactionError, BuildTransactionError } from '../baseCoin/errors';
 import { BaseAddress, BaseKey } from '../baseCoin/iface';
+import { isValidEd25519Seed } from '../../utils/crypto';
+import { testnetMetadataRpc, westendMetadataRpc, mainnetMetadataRpc } from './metadataRpc';
+import BigNumber from 'bignumber.js';
+import { decodeAddress } from '@polkadot/keyring';
+import { getRegistry, decode } from '@substrate/txwrapper-polkadot';
+import { UnsignedTransaction } from '@substrate/txwrapper-core';
+import { TypeRegistry, DecodedSignedTx, DecodedSigningPayload } from '@substrate/txwrapper-core/lib/types';
+import { BaseTransactionSchema, SignedTransactionSchema, SigningPayloadTransactionSchema } from './txnSchema';
 import { Transaction } from './transaction';
+import { KeyPair } from './keyPair';
+import { CreateBaseTxInfo, sequenceId, specNameType, TxMethod, Validity } from './iface';
+import Utils from './utils';
+import { AddressValidationError } from './errors';
 
 export abstract class TransactionBuilder extends BaseTransactionBuilder {
-  private _transaction: Transaction;
+  protected _transaction: Transaction;
+  protected _keyPair: KeyPair;
+  protected _sender: string;
+
+  protected _blockNumber: number;
+  protected _blockHash: string;
+  protected _genesisHash: string;
+  protected _metadataRpc: string;
+  protected _specVersion: number;
+  protected _transactionVersion: number;
+  protected _specName: specNameType;
+  protected _chainName: string;
+  protected _nonce: number;
+  protected _tip?: number;
+  protected _eraPeriod?: number;
+  protected _registry: TypeRegistry;
+  protected _method?: TxMethod;
 
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
+    this._transaction = new Transaction(_coinConfig);
+    this.staticsConfig();
+  }
+  /**
+   *
+   * Sets the address of sending account.
+   *
+   * @param {BaseAddress} address The SS58-encoded address.
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  sender({ address }: BaseAddress): this {
+    this.validateAddress({ address });
+    this._sender = address;
+    this._transaction.sender(address);
+    return this;
+  }
+  /**
+   *
+   * The number of blocks after the checkpoint for which a transaction is valid. If zero, the transaction is immortal.
+   *
+   * @param {number | undefined} maxDuration
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  durationConfig({ maxDuration }: { maxDuration: number | undefined }): this {
```

Comment:
I updated the doc to be part of the validity window method: https://docs.google.com/document/d/1cMGyXa8GwgwDMcZ2Uv7dMDsqRWlIEp1MRdVgpPW8p1o/edit#bookmark=id.6migwbru6vnd